python/ray/tests/test_placement_group.py

```diff
@@ -127,22 +100,63 @@ def value(self):
     assert node_of_actor_1 != node_of_actor_2
 
 
-@pytest.mark.skipif(
-    os.environ.get("RAY_GCS_ACTOR_SERVICE_ENABLED") != "true",
-    reason=("This edge case is not handled when GCS actor management is off. "
-            "We won't fix this because GCS actor management "
-            "will be on by default anyway."))
-def test_placement_group_spread_best_effort(ray_start_cluster):
-    @ray.remote(num_cpus=2)
-    class Actor(object):
-        def __init__(self):
-            self.n = 0
+def test_placement_group_actor_resource_ids(ray_start_cluster):
+    @ray.remote(num_cpus=1)
+    class F:
+        def f(self):
+            return ray.get_resource_ids()
 
-        def value(self):
-            return self.n
+    cluster = ray_start_cluster
+    num_nodes = 1
+    for _ in range(num_nodes):
+        cluster.add_node(num_cpus=4)
+    ray.init(address=cluster.address)
+
+    g1 = ray.experimental.placement_group([{"CPU": 2}])
+    a1 = F.options(placement_group_id=g1).remote()
+    resources = ray.get(a1.f.remote())
+    assert len(resources) == 1, resources
+    assert "_CPU" in list(resources.keys())[0], resources
+
+
+def test_placement_group_task_resource_ids(ray_start_cluster):
+    @ray.remote(num_cpus=1)
+    def f():
+        return ray.get_resource_ids()
+
+    cluster = ray_start_cluster
+    num_nodes = 1
+    for _ in range(num_nodes):
+        cluster.add_node(num_cpus=4)
+    ray.init(address=cluster.address)
+
+    g1 = ray.experimental.placement_group([{"CPU": 2}])
+    o1 = f.options(placement_group_id=g1).remote()
+    resources = ray.get(o1)
+    assert len(resources) == 1, resources
+    assert "_CPU" in list(resources.keys())[0], resources
+
+
+def test_placement_group_hang(ray_start_cluster):
+    @ray.remote(num_cpus=1)
+    def f():
+        return ray.get_resource_ids()
+
+    cluster = ray_start_cluster
+    num_nodes = 1
+    for _ in range(num_nodes):
+        cluster.add_node(num_cpus=4)
+    ray.init(address=cluster.address)
+
+    ray.get(f.remote())  # <--- TODO(ekl) this line causes the later hang
+
+    g1 = ray.experimental.placement_group([{"CPU": 2}])
+    o1 = f.options(placement_group_id=g1).remote()
 
-    # TODO(Shanly):
-    pass
+    # TODO(ekl) This hangs forever.
```

Comment:
@Alisahhh any idea why this test hangs? If I remove line 151, the hang no longer happens.